proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,484 @@
+  * [Node built-in importer](#node-built-in-importer)
+  * [Design Decisions](#design-decisions)
+    * [Using a `pkg:` URL scheme](#using-a-pkg-url-scheme)
+    * [No built-in `pkg:` resolver for browsers](#no-built-in-pkg-resolver-for-browsers)
+    * [Available as an opt-in setting](#available-as-an-opt-in-setting)
+    * [Available in legacy API](#available-in-legacy-api)
+    * [Node Resolution Decisions](#node-resolution-decisions)
+* [Types](#types)
+    * [`pkgImporter`](#pkgimporter)
+* [Semantics](#semantics)
+  * [Package Importers](#package-importers)
+  * [Node Package Importer](#node-package-importer)
+* [Procedures](#procedures)
+  * [Node Algorithm for Resolving a `pkg:` URL](#node-algorithm-for-resolving-a-pkg-url)
+  * [Resolving a package name](#resolving-a-package-name)
+  * [Resolving the root directory for a package](#resolving-the-root-directory-for-a-package)
+  * [Resolving package exports](#resolving-package-exports)
+  * [Resolving package root values](#resolving-package-root-values)
+  * [Export Load Paths](#export-load-paths)
+* [Ecosystem Notes](#ecosystem-notes)
+
+## Background
+
+> This section is non-normative.
+
+Historically, Sass has not specified a standard method for using packages from
+dependencies. A number of domain-specific solutions exist using custom importers
+or by specifying a load path. This can lead to Sass code being written in a way
+that is tied to a specific domain and make it difficult to rely on dependencies.
+
+## Summary
+
+> This section is non-normative.
+
+Sass users often need to use styles from a dependency to customize an existing
+theme or access styling utilities.
+
+This proposal defines a `pkg:` URL scheme for usage with `@use` that directs an
+implementation to resolve a URL within a dependency. The implementation will
+resolve the dependency URL using the standard resolution for that environment.
+Once resolved, this URL will be loaded in the same way as any other `file:` URL.
+
+This proposal also defines a built-in Node importer.
+
+For example, `@use "pkg:bootstrap";` would resolve to the path of a
+library-defined export within the `bootstrap` dependency. In Node, that would be
+resolved within `node_modules`, using the [Node resolution algorithm].
+
+[node resolution algorithm]: https://nodejs.org/api/packages.html
+
+### Node built-in importer
+
+The built-in Node importer resolves in the following order:
+
+1. `sass` condition in package.json `exports`.
+
+2. `style` condition in package.json `exports`.
+
+3. If no subpath, then find root export:
+
+  1. `sass` key at package.json root.
+
+  2. `style` key at package.json root.
+
+  3. `index` file at package root, resolved for file extensions and partials.
+
+4. If there is a subpath, resolve that path relative to the package root, and
+   resolve for file extensions and partials
+
+For library creators, the recommended method is to add a `sass` conditional
+export key as the first key in `package.json`.
```

Comment:
This is a great question. When possible, we'll be using `resolve.exports`, which always adds the 

There's a fair amount of conversation [here](https://github.com/lukeed/resolve.exports/issues/12), which boils down to wanting to adhere to the entirety of the node spec, which specifies:

- default
- import or require (The resolution spec we are referring to is specifically the ECMAScript Module, not CommonJS, so in our case it is import)
- node

I think we have a few options-

1. Explicitly pass `node`, `import`, and `default` in our algorithm to match the `resolve.exports` functionality 
2. Use `resolve.exports`'s `unsafe` mode which would only pass `default` in addition to our custom condition. This would break from default Node.js behavior
3. Adopt a simpler algorithm that would not follow Node.js behavior at all, but do a more direct lookup of `exports.[path].sass` (and many recursive variants). 

While I'm tempted by #3 in order to make things a bit simpler for users, I think the best decision is #1, as it best supports the ecosystem standards and conventions.

Reply:
As discussed offline, we can't use `resolve.exports` here directly because we'll need to implement the algorithm in pure Dart anyway (and all current implementations will be able to use that implementation directly). Also, since the Node spec itself doesn't require the use of `"node"` or `"import"` keys, we aren't bound by spec considerations to use them, so we'll only respect `"default"`, `"sass"`, and `"style"`.

Reply:
I updated this in 7f8ec04. Notably, I switched from a 2 step process that checked for `sass` and `style` separately to one that checks for both simultaneously. This addresses a case where the `style` and `default` conditions are set.